mne/preprocessing/maxwell.py

```diff
@@ -0,0 +1,530 @@
+                                               factorial(degree + order))) * \
+        -np.sin(pol) * _alegendre_deriv(degree, order, np.cos(pol)) * \
+        np.exp(1j * order * az)
+
+    # Get real component of vectors, convert to cartesian coords, and return
+    real_grads = _get_real_grad(np.c_[g_rad, g_az, g_pol], order)
+    return _sph_to_cart_partials(np.c_[rad, az, pol], real_grads)
+
+
+def _grad_out_components(degree, order, rad, az, pol):
+    """Compute gradient of external component of V(r) spherical expansion.
+
+    External component has form: Ylm(azimuth, polar) * (radius ** degree)
+
+    Parameters
+    ----------
+    degree : int
+        Degree of spherical harmonic. (Usually) corresponds to 'l'
+    order : int
+        Order of spherical harmonic. (Usually) corresponds to 'm'
+    rad : ndarray, shape (n_samples,)
+        Array of radii
+    az : ndarray, shape (n_samples,)
+        Array of azimuthal (longitudinal) spherical coordinates [0, 2*pi]. 0 is
+        aligned with x-axis.
+    pol : ndarray, shape (n_samples,)
+        Array of polar (or colatitudinal) spherical coordinates [0, pi]. 0 is
+        aligned with z-axis.
+
+    Returns
+    -------
+    ndarray, shape (n_samples, 3)
+        Gradient of the spherical harmonic and vector specified in rectangular
+        coordinates
+    """
+    # Compute gradients for all spherical coordinates (Eq. 7)
+    g_rad = degree * rad ** (degree - 1) * _sph_harmonic(degree, order, az,
+                                                         pol)
+
+    g_az = rad ** (degree - 1) / np.sin(pol) * 1j * order * \
+        _sph_harmonic(degree, order, az, pol)
+
+    g_pol = rad ** (degree - 1) * np.sqrt((2 * degree + 1) *
+                                          factorial(degree - order) /
+                                          (4 * np.pi *
+                                           factorial(degree + order))) * \
+        -np.sin(pol) * _alegendre_deriv(degree, order, np.cos(pol)) * \
+        np.exp(1j * order * az)
+
+    # Get real component of vectors, convert to cartesian coords, and return
+    real_grads = _get_real_grad(np.c_[g_rad, g_az, g_pol], order)
+    return _sph_to_cart_partials(np.c_[rad, az, pol], real_grads)
+
+
+def _get_real_grad(grad_vec_raw, order):
+    """Helper function to convert gradient vector to to real basis functions.
+
+    Parameters
+    ----------
+    grad_vec_raw : ndarray, shape (n_gradients, 3)
+        Gradient array with columns for radius, azimuth, polar points
+    order : int
+        Order (usually 'm') of multipolar moment.
+
+    Returns
+    -------
+    grad_vec : ndarray, shape (n_gradients, 3)
+        Gradient vectors with only real componnet
+    """
+
+    if order > 0:
+        grad_vec = np.sqrt(2) * np.real(grad_vec_raw)
+    elif order < 0:
+        grad_vec = np.sqrt(2) * np.imag(grad_vec_raw)
+    else:
+        grad_vec = grad_vec_raw
+
+    return np.real_if_close(grad_vec)
```

Comment:
why use `real_if_close` here? Can we guarantee that it's real instead? `np.real` and `np.imag` in the first two conditionals will give us real values, so I assume this has to do with the last bit. I think you can just change that to `grad_vec = grad_vec_raw.real`, because you have coded this to be real (right)? Or are there actually cases where you want it to return `np.complex128` data?